src/cider/nrepl/middleware/content_type.clj

```diff
@@ -0,0 +1,49 @@
+(ns cider.nrepl.middleware.content-type
```

Comment:
I'd add some high-level documentation of the ns here (and maybe list the shortcomings/design decisions listed in the PR description).